typed-racket-more/typed/rackunit/main.rkt

```diff
@@ -93,22 +94,108 @@
   [t:check-around check-around]
   [t:current-check-handler current-check-handler]
   [t:current-check-around current-check-around])
-(define-rewriter t:test-case test-case
-  [t:current-test-case-around current-test-case-around]
-  [t:check-around check-around]
-  [t:current-check-handler current-check-handler]
-  [t:current-check-around current-check-around])
+
+(define-syntax-rule (test-case name expr ...)
+  (parameterize
+      ([current-test-name (ann name String)])
+    (test-begin expr ...)))
+
 (provide test-begin test-case)
 
 (require/opaque-type TestCase test-case? rackunit)
 (provide TestCase test-case?)
 
+(require/typed
+ rackunit/private/monad
+ [#:opaque monad monad?])
+(define-type Seed (Option monad))
 
+(define-type test-suite-handler-down
+  (rackunit-test-suite (Option String) (Thunk Any) (Thunk Any) Seed -> Seed))
```

Comment:
Do these Thunks need to accomodate returning multiple values -- ie, using `AnyValues` instead of `Any`?